xstate-chatbot/nodejs/src/machine/vitals.js

```diff
@@ -18,11 +18,54 @@ const vitalsFlow = {
       id: 'rrtSrfId',
       initial: 'prompt',
       states: {
-        prompt: {
+       prompt: {
+         onEntry: assign((context, event) => {
+           dialog.sendMessage(context, dialog.get_message(messages.rrtSrfId.prompt, context.user.locale));
+          }),
+         on: {
+            USER_MESSAGE: 'process'
+          }
+        },
+        process: {
           onEntry: assign((context, event) => {
-            dialog.sendMessage(context, dialog.get_message(messages.rrtSrfId.prompt, context.user.locale));
+            context.message = dialog.get_input(event, false);
           }),
-          on: {
+          invoke: {
+            src: (context, event) => vitalsService.getPatientDetailsFromSrfId(context.message),
+            onDone: [
+              {
+                cond: (context, event) => event.data.response === 1,
+                actions: assign((context, event) => {
+                  context.slots.vitals.srfId = context.message;
+                  context.slots.vitals.mobile_no = event.data.data[0].mobile_no;
+                  context.message = undefined;
+                  dialog.sendMessage(context, dialog.get_message(messages.rrtSrfId.success, context.user.locale), false);
+                }),
+                target: '#temperature'
+              },
+              {
+                target: 'error'
+              }
+            ]
+          }
+        },
+        error: {
+          onEntry: assign((context, event) => {
+            dialog.sendMessage(context, dialog.get_message(messages.rrtSrfId.error), false);
+          }),
+          always: 'prompt'
+        }
+     }
+    },
+    patrmo: {
```

Comment:
Please write meaningful names. This name is going to show up on dashboard

Reply:
patrmo (patient through RMO) state will be used for  RMO Flow which is not yet introduced but will changes its name while adding RMO flow.

Reply:
Only commit when the chat flow is ready.
I suggest you could name the state as "patientThroughRMO" instead of "pattmo". It is easier to understand.